src/zenml/config/step_configurations.py

```diff
@@ -69,6 +69,7 @@ class StepConfigurationUpdate(StrictBaseModel):
     parameters: Dict[str, Any] = {}
     settings: Dict[str, BaseSettings] = {}
     extra: Dict[str, Any] = {}
+    on_failure: Optional[str] = None
```

Comment:
```suggestion
    failure_hook_source: Optional[str] = None
```
seems better to me, also needs some changing elsewhere